### PR TITLE
[GNA] fix InsertCopyLayerPass

### PR DESCRIPTION
**Pull request by kmagiers**

### CODE REVIEW

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 1]
```diff
@@ -702,50 +702,52 @@ void InsertIdentityLayerPass::run() {
...
+                    auto concatLayer = ConcatLayers[concatIdx].first;
+                    auto inputIdx = ConcatLayers[concatIdx].second;
+                    InsertCopyLayer(l, concatLayer, inputIdx, this->getPassManager(), CopyLayerName);
                 }
-                auto prevLayer = CNNNetPrevLayer(l, i);
-                InsertCopyLayer(prevLayer, l, i, getPassManager(), bInsertDelayed ? DelayedCopyLayerName : CopyLayerName);
+                currentCopyIdx++;
             }
         }
     }
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:753]: Could you please add a simple test for it?

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 2]
```diff
@@ -702,50 +702,52 @@ void InsertIdentityLayerPass::run() {
...
+        for (auto output: l->outData) {
+            std::vector<std::pair<CNNLayerPtr, size_t>> MemoryLayers;
+            std::vector<std::pair<CNNLayerPtr, size_t>> ConcatLayers;
+            auto& inputTo = getInputTo(output);
+            if (inputTo.size() < 2) continue;
+            for (auto& childLayer : inputTo) {
+                auto layer_to_insert = childLayer.second;
+                auto current_layer = childLayer.second;
+                auto previous_layer = l;
+                size_t input_idx = CNNLayerFindInsDataIdxes(output, current_layer)[0];
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:715]: It is still good to check if a list is not empty and take first element

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 3]
```diff
@@ -702,50 +702,52 @@ void InsertIdentityLayerPass::run() {
...
                 }
             }
-            if (!isInserted() && LayerInfo(l).isConcat() && LayerInfo(prevIndirectLayer).isCrop()) { bInsert = true; }
-
-            if (isInserted()) {
-                if (LayerInfo(prevIndirectLayer).isCropAffined()) {
-                    // The crop will be replaced by affine.
-                    // Copy layer insertion is not required
-                    continue;
+            if (MemoryLayers.size() == 0 && ConcatLayers.size() == 0) continue;
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:735]: MemoryLayers.empty() && ConcatLayers.empty()

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp (hunk 2 above)
Comment (kmagiers) [reply inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:787]: CNNLayerFindInsDataIdxes asserts it is not empty

**Review by bsochack — APPROVED** [review]

**Pull request merged**